llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -4220,8 +4220,9 @@ bool llvm::canReplaceOperandWithVariable(const Instruction *I, unsigned OpIdx) {
     return false;
 
   // Early exit.
-  if (!isa<Constant>(I->getOperand(OpIdx)))
+  if (!isa<Constant, InlineAsm>(I->getOperand(OpIdx))) {
     return true;
+  }
```

Comment:
I'll fix the bracing here after I see the CI turn green on Windows. I already have the fix locally.

Reply:
Shoot of course I forgot to update before hitting merge.. addressed in #138620